### ISSUE TITLE
Consider Switching to Per-User Access Instead of the single "Authors" group

### ISSUE THREAD

**Issue by quicksketch** [backdrop-ops/contrib#150]
When we set up the contrib group, GitHub required that commit access to a repository be granted to a *group*, and then users in turn could be assigned to that group.

More recently [GitHub expanded their permissions system](https://github.com/blog/2020-improved-organization-permissions), that allowed users to be given direct commit access to an individual project, with no group being necessary.

We might consider switching to this model where commit access is per-project, rather than all-or-nothing commit access to *every* project. Let's look into switching our implementation and what sort of effect it would have on our process, as well as if any new tools are needed.

**Comment by klonos** [comment]
Agreed. People might be more used to the d.org model anyways where users have commit access only if they are either maintainers or co-maintainers of the project. It makes more sense I guess. But we still need a way to be able to allow access to future contributors should a project be either neglected or officially abandoned.

We will have to change our official agreement of acceptance too.

How will this work for people like myself? You see, I like to contribute by triaging the issue queue and this often requires permission to edit issue summaries (even if created by others - no matter which queue it belongs to). Think meta-issues where tasks need to be ticked off as they get done and more added as new sub-tasks are decided. Also a recent case where a security issue was publicly posted and I had to edit/remove the comment. Things like that.

**Comment by biolithic** [comment]
I like the group model in that I might write something wrong in the readme and then a mod can edit it, or I might go away for a reason and then the code belongs to Backdrop rather than being abandoned.  Or "things have to be ported like Webform and Services and etc", but I don't necessarily want to be the maintainer for everything, so it would be nice to hand something off easily.

**Comment by quicksketch** [comment]
Hm, I just realized that access to labels is still restricted to committers only. So the group model helps organize issues as well.

**Comment by sentaidigital** [comment]
As long as the owners of backdrop-contrib have the ability to add / remove committers to an individual repository, 

> How will this work for people like myself? You see, I like to contribute by triaging the issue queue and ...

GitHub has a very small number of roles, probably because they expect anyone who has write access to be a fully trusted member of the team.

**Comment by sentaidigital** [comment]
As long as the backdrop-contrib owners can add contributors to an individual repository, I'm not worried about abandoned projects.  The hard part is finding someone willing to take it over.

> How will this work for people like myself? You see, I like to contribute by triaging the issue queue and this often requires permission to...

GitHub has a very small number of roles, probably because they expect anyone with write access to any part is a fully trusted member of the team and the binary read-only / read-write is easy to understand.  I would like to see GitHub add roles like "editor" (edit wiki pages, maybe issues) and "scrum master" (full admin over the issue queue, no commit access.)  Maybe these are (or could be) available in the non-free tiers.

**Comment by jenlampton** [comment]
I'd also like to stick to the group-access model, since that will enable the "bug squad" we talked about earlier to manage issues on behalf of busy maintainers. I also like the feeling of "we're all in this together" - it feels like we're more of a community this way :)

**Comment by alexfinnarn** [comment]
can there be a compromise between per-repo permissions and teams/groups? The per-project permissions are handled automatically by a bot and then the team/group members are tapped/coronated by more involved/long-time contributors?

**Comment by laryn** [comment]
I agree that the "any contributor can commit to any contrib project" set up makes me a little nervous. Not necessarily because of nefarious reasons (like someone gets hacked and the whole of contrib is open to the hacker), but even because of the possibility of accidental commits in someone else's repo. Is there a way to allow the Security Team (and/or Bug Squad group) to automatically have access to all repos but then have a per-user control as well? That way everything is always within admin control but individual users could be (and add) maintainers.

**Comment by jenlampton** [comment]
##Roles

* The contrib group has 3 levels of global access: **Admin**, **Write**, **Read**, (and None)
* Repositories have 5 levels of access: **Admin**, **Maintain**, **Write**, **Triage**, **Read** (and None)

##Permissions 

There are 3 teams that should have access to all repositories. Base permissions to all repositories apply to all members 

* Base permissions - currently has/should have **Write** access to all repos
* Authors - currently has/should have **Write** access to all repos
* Bug Squad - currently has/should have **Write** access to all repos
* Security Team - currently has/should have **Admin** access to all repos

##Repositories

In addition to the above, permissions should be set up as follows for each individual repository:

* Owner - currently has/should have **Admin** access to this repo only
* Maintainers (optional) - could be specifically added by owner at any level


##Questions:

1) Should we require 2FA for all backdrop-contrib gruop members?
2) Should we change Base permissions to **None**? 
3) Should we change Authors team permissions to **Triage**?

**Comment by jenlampton** [comment]
There are two very important things that people with **Maintain** access are not allowed to do to their repos:
* change the default branch
* Add other maintaoners (collaborators)

Because of this, I think it's important that we allow some people **Admin** access to their own projects. 

Anyone with Admin access to any Backdrop repo should be required to have 2FA enabled for their GitHub account (this includes everyone on the security team, but I expect they all have it already)
